File: mywebsite-parent/mywebsite-content-interface/src/main/java/cn/boom/mywebsite/content/service/CommentService.java
package cn.boom.mywebsite.content.service;

import cn.boom.mywebsite.pojo.MyWebSiteResult;
import cn.boom.mywebsite.pojo.TbComment;

public interface CommentService {

    /**
     * 根据id查询评论
     * @param id
     * @return
     */
    public abstract TbComment findOneById(Long id);

    /**
     * 条件分页搜索
     * @param tbComment
     * @param pageNum
     * @param pageSize
     * @return
     */
    public abstract MyWebSiteResult search(TbComment tbComment, int pageNum, int pageSize);


    /**
     * 添加评论
     * @param tbComment
     * @return
     */
    public abstract MyWebSiteResult add(TbComment tbComment);

    /**
     * 批量审核
     * @param ids
     * @param status
     * @return
     */
    public abstract MyWebSiteResult updateStatusIds(Long[] ids, String status);

    /**
     * 批量删除（关联删除）
     * @param ids
     * @return
     */
    public abstract MyWebSiteResult deleteIds(Long[] ids);

    /**
     * 修改评论信息
     * @param tbComment
     * @return
     */
    public abstract MyWebSiteResult update(TbComment tbComment);

    /**
     * 通过博客id查询所有评论
     * @param blogId
     * @return
     */
    public abstract MyWebSiteResult getCommentByBlogId(Long blogId);

    /**
     * 通过动态id查询所有评论
     * @param trendsId
     * @return
     */
    public abstract MyWebSiteResult getCommentByTrendsId(Long trendsId);

    /**
     * 点赞
     * @param id
     * @return
     */
    public abstract MyWebSiteResult updateCommentPraseCount(Long id);
}

File: mywebsite-parent/mywebsite-portal-web/src/main/webapp/angularjs/js/service/commentService.js
app.service('commentService', function ($http) {

    /**
     * 条件分页搜索
     * @param searchEntity
     * @param page
     * @param rows
     * @returns {*}
     */
    this.search = function (searchEntity, page, rows) {
        return $http.post('../comment/search.do?page=' + page + '&rows=' + rows, searchEntity);
    };

    /**
     * 根据id查询评论
     * @param id
     * @returns {*}
     */
    this.findOneById = function (id) {
        return $http.get('../comment/findOneById.do?id=' + id);
    };

    /**
     * 更新评论
     * @param entity
     * @returns {*}
     */
    this.update = function (entity) {
        return $http.post('../comment/update.do', entity);
    };

    /**
     * @param entity
     * @returns {*}
     */
    this.add = function (entity) {
        return $http.post('../comment/add.do', entity);
    };

    /**
     * 批量删除
     * @param ids
     * @returns {*}
     */
    this.deleteIds = function (ids) {
        return $http.get('../comment/deleteIds.do?ids='+ids);
    };

    /**
     * 批量审核
     * @param ids
     * @param status
     * @returns {*}
     */
    this.updateStatusIds = function (ids, status) {
        return $http.get('../comment/updateStatusIds.do?ids=' + ids + '&status=' + status);
    };

    /**
     * 根据博客id查询评论信息
     * @param blogId
     * @returns {*}
     */
    this.getCommentByBlogId = function (blogId) {
        return $http.get('../comment/getCommentByBlogId.do?blogId=' + blogId);
    };

    /**
     * 根据动态id查询评论信息
     * @param trendsId
     * @returns {*}
     */
    this.getCommentByTrendsId = function (trendsId) {
        return $http.get('../comment/getCommentByTrendsId.do?trendsId=' + trendsId);
    };

    /**
     * 点赞
     * @param id
     * @returns {*}
     */
    this.updateCommentPraseCount = function (id) {
        return $http.get('../comment/updateCommentPraseCount.do?id=' + id);
    };
});